File: example/src/javaExample/java/com/bugsnag/android/example/ExampleApplication.java
package com.bugsnag.android.example;

import android.app.Application;

import com.bugsnag.android.Bugsnag;
import com.bugsnag.android.Configuration;

public class ExampleApplication extends Application {

    @Override
    public void onCreate() {
        super.onCreate();
        Configuration config = new Configuration("f35a2472bd230ac0ab0f52715bbdc65d");
        config.setSessionEndpoint("http://10.0.2.2:8092");
        config.setEndpoint("http://10.0.2.2:8092");

        Bugsnag.init(this, config);
        Bugsnag.setAutoCaptureSessions(true);

    }

}

File: examples/sdk-app-example/src/main/java/com.bugsnag.android/example/ExampleApplication.kt
package com.bugsnag.android.example

import android.app.Application
import com.bugsnag.android.Bugsnag
import com.bugsnag.android.Configuration
import com.bugsnag.android.ErrorTypes

class ExampleApplication : Application() {

    companion object {
        init {
            System.loadLibrary("entrypoint")
        }
    }

    private external fun performNativeBugsnagSetup()

    override fun onCreate() {
        super.onCreate()

        val config = Configuration.load(this)
        config.enabledErrorTypes.ndkCrashes = true
        config.setUser("123456", "joebloggs@example.com", "Joe Bloggs")
        config.addMetadata("user", "age", 31)
        Bugsnag.start(this, config)

        // Initialise native callbacks
        performNativeBugsnagSetup()
    }

}
